yt_dlp/extractor/bilibili.py

```diff
@@ -861,7 +866,7 @@ def _real_extract(self, url):
         headers['Referer'] = url
         play_info = self._download_json(
             'https://api.bilibili.com/pgc/player/web/v2/playurl', episode_id,
-            'Extracting episode', query={'fnval': '4048', 'ep_id': episode_id},
+            'Extracting episode', query={'fnval': 12240, 'ep_id': episode_id},
             headers=headers)
```

Comment:
For bangumi like `https://www.bilibili.com/bangumi/play/ep829434`, all logged-in users can access premium quality, and  all qualities are directly embedded in `__playinfo__` of page HTML like non-festival `BiliBiliIE` videos. (The trial counter can be simply ignored). May be worth taking a look at that or see if we can tweak API request query to directly achieve the same.

Reply:
I was reading and testing the js and found that by changing `this.state.reqQuality` to 120 in [`core.8f2238a5.js`](https://s1.hdslb.com/bfs/static/player/main/core.8f2238a5.js) on https://www.bilibili.com/bangumi/play/ep829434, we can access the quality `120` stored in the `__playinfo__`(the 4k quality).
Not sure if we can do this on other bangumi.

Reply:
let's leave this to another pr